src/main/java/ch/njol/skript/effects/EffStructureSaveUnregister.java

```diff
@@ -0,0 +1,89 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import java.io.IOException;
+
+import org.bukkit.Bukkit;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.structure.StructureManager;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+
+@Name("Structure Save/Unregister")
+@Description("Unregisters or saves a structure by it's namespace key.")
+@Examples("unregister structure named \"Example\"")
+@RequiredPlugins("Minecraft 1.17.1+")
```

Comment:
Which part of this is specific to 1.17.1?
Structures were added in Minecraft [1.13](https://minecraft.fandom.com/wiki/Structure_Block) and the Spigot API has supported them since 1.14.

Reply:
You're looking at the physical Structure Block class. Spigot only added an API to utilize the backend of how Mojang uses Structures here https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/6be36d3638f97e6e0dc053f3e0fd0e68e8ec93f0